numpy/lib/format.pyi

```diff
@@ -20,3 +20,6 @@ def read_array_header_2_0(fp): ...
 def write_array(fp, array, version=..., allow_pickle=..., pickle_kwargs=...): ...
 def read_array(fp, allow_pickle=..., pickle_kwargs=...): ...
 def open_memmap(filename, mode=..., dtype=..., shape=..., fortran_order=..., version=...): ...
+class AppendArray:
+    def __init__(self, filename): None
+    def append(self, arr): None
```

Comment:
`def func(self): None` is not a valid expression, so this will have to be updated:
```diff
- class AppendArray:
-     def __init__(self, filename): None
-     def append(self, arr): None
+ import os
+ from numpy.typing import NDArray
+
+ class AppendArray:
+     def __init__(self, filename: str | bytes | os.PathLike[str | bytes]) -> None: ...
+     def append(self, arr: NDArray[Any]) -> None: ...
```

Reply:
Secondly, a number of (public) methods and attributes are still absent from the stub file (`__enter__`, `fp`, _etc._),
can add those as well?

Reply:
Yes, that would be great, I have simply forgotten it in a hurry. Looks much nicer (and useful) with type annotations as well. Does it make sense to make `fp` public? Does that fit with the numpy style? Otherwise I'd simply make everything private but special methods, `append` and `close`. How does the modification work technically? Will it change the pull request? Can I pull it back into my forked repository?

Reply:
I think I've figured out, this is just a comment with diff syntax highlighting. Just changed it in my repo.